Panic when DynamicKubeletConfig is enabled.

Feedback from @luxas on slack.

<details>

```
_output/bin/kubeadm init --feature-gates DynamicKubeletConfig=true --kubernetes-version ci/latest
[kubeadm] WARNING: kubeadm is currently in beta
[init] Using Kubernetes version: v1.10.0-alpha.0.100+2d972c19bfdf2e
[init] Using Authorization modes: [Node RBAC]
[preflight] Running pre-flight checks.
[preflight] WARNING: kubernetes version is greater than kubeadm version. Please consider to upgrade kubeadm. kubernetes version: 1.10.0-alpha.0.100+2d972c19bfdf2e. Kubeadm version: 1.9.x
[preflight] WARNING: crictl not found in system path
[preflight] WARNING: docker service is not enabled, please run 'systemctl enable docker.service'
[preflight] Starting the kubelet service
[certificates] Generated ca certificate and key.
[certificates] Generated apiserver certificate and key.
[certificates] apiserver serving cert is signed for DNS names [thegopher kubernetes kubernetes.default kubernetes.default.svc kubernetes.default.svc.cluster.local] and IPs [10.96.0.1 192.168.1.115]
[certificates] Generated apiserver-kubelet-client certificate and key.
[certificates] Generated sa key and public key.
[certificates] Generated front-proxy-ca certificate and key.
[certificates] Generated front-proxy-client certificate and key.
[certificates] Valid certificates and keys now exist in "/etc/kubernetes/pki"
[kubeconfig] Wrote KubeConfig file to disk: "admin.conf"
[kubeconfig] Wrote KubeConfig file to disk: "kubelet.conf"
[kubeconfig] Wrote KubeConfig file to disk: "controller-manager.conf"
[kubeconfig] Wrote KubeConfig file to disk: "scheduler.conf"
[controlplane] Wrote Static Pod manifest for component kube-apiserver to "/etc/kubernetes/manifests/kube-apiserver.yaml"
[controlplane] Wrote Static Pod manifest for component kube-controller-manager to "/etc/kubernetes/manifests/kube-controller-manager.yaml"
[controlplane] Wrote Static Pod manifest for component kube-scheduler to "/etc/kubernetes/manifests/kube-scheduler.yaml"
[etcd] Wrote Static Pod manifest for a local etcd instance to "/etc/kubernetes/manifests/etcd.yaml"
[init] Waiting for the kubelet to boot up the control plane as Static Pods from directory "/etc/kubernetes/manifests".
[init] This might take a minute or longer if the control plane images have to be pulled.
[apiclient] All control plane components are healthy after 53.002568 seconds
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x29ab0bd]

goroutine 1 [running]:
k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet.UpdateNodeWithConfigMap.func1(0xc420cb2060, 0xc420ee3727, 0xc420ee3701)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet/kubelet.go:89 +0x20d
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.WaitFor(0xc420b4c280, 0xc420b400f0, 0xc420cb2000, 0x8d186d, 0x31dcfa0)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:311 +0x6b
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.pollInternal(0xc420b4c280, 0xc420b400f0, 0x0, 0x0)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:227 +0x94
k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait.Poll(0x1dcd6500, 0x1bf08eb000, 0xc420b400f0, 0x0, 0x0)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go:221 +0x4d
k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet.UpdateNodeWithConfigMap(0x52e9260, 0xc420867c00, 0xc4207a8290, 0x9, 0x0, 0x820)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet/kubelet.go:73 +0xaa
k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet.CreateBaseKubeletConfiguration(0xc4203b0000, 0x52e9260, 0xc420867c00, 0x14, 0xc420a8c9e8)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/app/phases/kubelet/kubelet.go:67 +0x3ac
k8s.io/kubernetes/cmd/kubeadm/app/cmd.(*Init).Run(0xc4201a03f0, 0x52b0d80, 0xc42000c018, 0x0, 0x36aa94f)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/init.go:360 +0x1bb9
k8s.io/kubernetes/cmd/kubeadm/app/cmd.NewCmdInit.func1(0xc4202186c0, 0xc4207bfec0, 0x0, 0x4)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/app/cmd/init.go:132 +0x24e
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).execute(0xc4202186c0, 0xc4207bfe80, 0x4, 0x4, 0xc4202186c0, 0xc4207bfe80)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:603 +0x234
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xc420280fc0, 0xc420210900, 0xc420219440, 0xc4202186c0)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:689 +0x2fe
k8s.io/kubernetes/vendor/github.com/spf13/cobra.(*Command).Execute(0xc420280fc0, 0xc42000c010, 0x52b0d80)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/vendor/github.com/spf13/cobra/command.go:648 +0x2b
k8s.io/kubernetes/cmd/kubeadm/app.Run(0xc4209bff70, 0x4046f4)
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/app/kubeadm.go:36 +0xe3
main.main()
	/home/lucas/luxas/kubernetes/_output/local/go/src/k8s.io/kubernetes/cmd/kubeadm/kubeadm.go:27 +0x26
```

</details>

--- a/cmd/kubeadm/app/phases/kubelet/kubelet.go
+++ b/cmd/kubeadm/app/phases/kubelet/kubelet.go
@@ -86,7 +86,13 @@ func UpdateNodeWithConfigMap(client clientset.Interface, nodeName string) error 
 			return false, nil
 		}
 
-		node.Spec.ConfigSource.ConfigMapRef.UID = kubeletCfg.UID
+		node.Spec.ConfigSource = &v1.NodeConfigSource{
+			ConfigMapRef: &v1.ObjectReference{
+				Name:      kubeadmconstants.KubeletBaseConfigurationConfigMap,
+				Namespace: metav1.NamespaceSystem,
+				UID:       kubeletCfg.UID,
+			},
+		}
 
 		newData, err := json.Marshal(node)
 		if err != nil {

